### ISSUE TITLE
Boost 1.60

### ISSUE THREAD

**Issue by bob-carpenter** [stan-dev/math#221]
The Boost headers (BH) package for R has updated, so we need to update for Stan for the next release.

**Comment by bgoodri** [comment]
Boost 1.6 deprecates one of the headers the parser was using
```
                 from lang__grammars__functions_grammar_inst.cpp:18:
/opt/r-devel/library/BH/include/boost/type_traits/detail/bool_trait_def.hpp:18:79: note: #pragma message: NOTE: Use of this header (bool_trait_def.hpp) is deprecated
 # pragma message("NOTE: Use of this header (bool_trait_def.hpp) is deprecated")
```

**Comment by bgoodri** [comment]
Also template_arity_spec.hpp
```
                 from /opt/r-devel/library/StanHeaders/include/src/stan/lang/parser.hpp:9,
                 from /opt/r-devel/library/StanHeaders/include/src/stan/lang/compiler.hpp:6,
                 from stanc.cpp:22:
/opt/r-devel/library/BH/include/boost/type_traits/detail/template_arity_spec.hpp:13:84: note: #pragma message: NOTE: Use of this header (template_arity_spec.hpp) is deprecated
 # pragma message("NOTE: Use of this header (template_arity_spec.hpp) is deprecated")
```

**Comment by bob-carpenter** [comment]
Thanks.  I'm glad that's all that came up.  I assigned this to myself and I'll try to fix those warnings.

Is there a branch where you merged in 1.60?  I didn't see one pushed to GitHub.

**Comment by bgoodri** [comment]
No brach. The tests just passed, albeit with more warnings when I put BH
1.60 on the disk.
